app/routes/session.py

```diff
@@ -139,23 +141,33 @@ def login() -> Response:
 
 
 def _set_questionnaire_supplementary_data(
-    questionnaire_store: QuestionnaireStore, metadata: MetadataProxy
+    *,
+    questionnaire_store: QuestionnaireStore,
+    metadata: MetadataProxy,
+    schema: QuestionnaireSchema,
 ) -> None:
     """
     If the survey metadata has an sds dataset id, and it either doesn't match what it stored, or there is no stored supplementary data
-    then fetch it and add it to the store
-    """
-    if not (new_sds_dataset_id := metadata["sds_dataset_id"]):
-        return
+    then fetch it, verify any schema supplementary lists are included in the fetched data, and add it to the questionnaire store
 
+    Validation of the supplementary lists must be performed every time a survey launches, not just when the supplementary data is fetched
+    as it is possible that the survey has changed but the dataset hasn't so the validity could have changed.
+    """
     existing_sds_dataset_id = (
         questionnaire_store.metadata.survey_metadata["sds_dataset_id"]
         if questionnaire_store.metadata and questionnaire_store.metadata.survey_metadata
         else None
     )
 
-    if existing_sds_dataset_id == new_sds_dataset_id:
-        # no need to fetch again
+    if (
+        not (new_sds_dataset_id := metadata["sds_dataset_id"])
+        or existing_sds_dataset_id == new_sds_dataset_id
+    ):
+        # no need to fetch: either no supplementary data or it hasn't changed, just validate lists
+        _validate_supplementary_data_lists(
```

Comment:
Longer term we can check if the cir instrument id changed as well (i.e schema has changed)